src/shared/dissect-image.c

```diff
@@ -260,24 +260,20 @@ static int make_partition_devname(
 
         return asprintf(ret, "%s%s%i", whole_devname, need_p ? "p" : "", nr);
 }
-#endif
 
-int dissect_image(
+static int dissect_image(
                 int fd,
                 const char *devname,
-                const char *image_path,
                 const VeritySettings *verity,
                 const MountOptions *mount_options,
                 DissectImageFlags flags,
-                DissectedImage **ret) {
+                DissectedImage *m) {
```

Comment:
should probably move to the front of the arguments list now, given it's the primary object we operate on that we take as input